overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -1,78 +1,409 @@
 import itertools, copy
```

Comment:
It's hard for me to tell which changes you made and which changes Nathan made in this file, so I haven't looked at it in depth. Let me know if there's anything you wanted me to look at more closely!

Reply:
Nathan and I have kind of gone on different paths on this file. I will be spending a lot of time merging this file.